File: app/Models/TherapyGoal.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Database\Eloquent\Relations\HasOne;
use Illuminate\Database\Eloquent\SoftDeletes;

class TherapyGoal extends Model
{
    use SoftDeletes;

    public const STATUSES = ['active', 'met', 'dropped'];

    /** Goal Attainment Scaling labels, keyed by rating value (-2…+2). */
    public const RATING_LABELS = [
        -2 => 'Much less than expected',
        -1 => 'Somewhat less than expected',
        0 => 'Expected level',
        1 => 'Somewhat more than expected',
        2 => 'Much more than expected',
    ];

    protected $fillable = [
        'patient_id',
        'clinician_id',
        'description',
        'status',
        'target_date',
    ];

    protected function casts(): array
    {
        return [
            'target_date' => 'date',
        ];
    }

    public function patient(): BelongsTo
    {
        return $this->belongsTo(Patient::class);
    }

    public function clinician(): BelongsTo
    {
        return $this->belongsTo(Clinician::class);
    }

    public function ratings(): HasMany
    {
        return $this->hasMany(GoalRating::class);
    }

    /** Most recent GAS rating for the goal. */
    public function latestRating(): HasOne
    {
        return $this->hasOne(GoalRating::class)->latestOfMany();
    }
}

File: theraconnect_flutter/lib/models/therapy_goal.dart
/// A therapy goal co-defined by the clinician, optionally with the most recent
/// Goal Attainment Scaling (GAS) rating. Read-only on the patient app.
class TherapyGoal {
  final int id;
  final String description;
  final String status; // 'active' | 'met'
  final String? targetDate;
  final GoalRatingSummary? latestRating;

  const TherapyGoal({
    required this.id,
    required this.description,
    required this.status,
    this.targetDate,
    this.latestRating,
  });

  factory TherapyGoal.fromJson(Map<String, dynamic> json) {
    return TherapyGoal(
      id: json['id'] as int,
      description: json['description'] as String,
      status: json['status'] as String,
      targetDate: json['target_date'] as String?,
      latestRating: json['latest_rating'] is Map<String, dynamic>
          ? GoalRatingSummary.fromJson(json['latest_rating'] as Map<String, dynamic>)
          : null,
    );
  }
}

class GoalRatingSummary {
  final int rating; // -2 … +2
  final String? label;
  final String? note;
  final String? createdAt;

  const GoalRatingSummary({
    required this.rating,
    this.label,
    this.note,
    this.createdAt,
  });

  factory GoalRatingSummary.fromJson(Map<String, dynamic> json) {
    return GoalRatingSummary(
      rating: (json['rating'] as num).toInt(),
      label: json['label'] as String?,
      note: json['note'] as String?,
      createdAt: json['created_at'] as String?,
    );
  }
}
